File: 2/2540/MinimumCommonValue.cpp
/*
https://leetcode.com/problems/minimum-common-value/description/?envType=daily-question&envId=2024-03-09
*/

class Solution {
public:
    int getCommon(vector<int>& nums1, vector<int>& nums2) {
        unordered_map<int, int>mp;
        for(auto &vals: nums1){
            mp[vals]++;
        }
        for(auto &vals: nums2){
            if(mp.find(vals)!=mp.end()) return vals;
        }
        return -1;
    }
};

File: 2/2000s/25s/2540/MinimumCommonValue.js
/*
https://leetcode.com/problems/minimum-common-value/description/?envType=daily-question&envId=2024-03-09
*/

/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number}
 */
var getCommon = function(nums1, nums2) {
    let mp=new Map();
    let sz=nums1.length, sz2=nums2.length;
    for(let i=0; i<sz; ++i){
        if(mp.has(nums1[i])){
            mp.set(nums1[i], mp.get(nums1[i])+1);
        } else mp.set(nums1[i], 1);
    }
    for(let i=0; i<sz2; ++i){
        if(mp.has(nums2[i])) return nums2[i];
    }
    return -1;
};